### PR TITLE
Fix failing tests after PR clash

### PR DESCRIPTION
**Pull request by craigtaverner**
Two PRs conflicted without github or CI noticing. The first added these tests, and the second modified their behaviour. Both went green in CI and both were merged within an hour of each other.

* PR that added the tests:
  * https://github.com/elastic/elasticsearch/pull/112938
  * merged 14:13CET
* PR that changed the behaviour of these tests:
  * https://github.com/elastic/elasticsearch/pull/114411
  * merged 14:48CET

### CODE REVIEW

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/PhysicalPlanOptimizerTests.java [hunk 1]
```diff
@@ -3243,23 +3243,21 @@ public void testPushWhereEvalToSource() {
...
         exchange = as(topLimit.child(), ExchangeExec.class);
         var project = as(exchange.child(), ProjectExec.class);
         var fieldExtract = as(project.child(), FieldExtractExec.class);
         assertThat(fieldExtract.attributesToExtract().size(), greaterThan(5));
-        limit = as(fieldExtract.child(), LimitExec.class);
-        var eval = as(limit.child(), EvalExec.class);
-        fieldExtract = as(eval.child(), FieldExtractExec.class);
-        assertThat(fieldExtract.attributesToExtract().stream().map(Attribute::name).collect(Collectors.toList()), contains("scalerank"));
         var source = source(fieldExtract.child());
         var condition = as(source.query(), SingleValueQuery.Builder.class);
```
Comment (alex-spies) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/PhysicalPlanOptimizerTests.java:3262]: nit: We could assert that the limit is now in the EsQueryExec for completeness in both tests.

**Review by alex-spies — APPROVED** [review]
Thanks @craigtaverner for taking care of our PRs' race!

**Comment by craigtaverner** [comment]
Fixes https://github.com/elastic/elasticsearch/issues/114628
Fixes https://github.com/elastic/elasticsearch/issues/114627

**Comment by craigtaverner** [comment]
Backported to 8.16 in https://github.com/elastic/elasticsearch/pull/114707

**Pull request merged**